docs/DeveloperGuide.adoc

```diff
@@ -114,99 +186,428 @@ NOTE: The lifeline for `DeleteCommandParser` should end at the destroy marker (X
+As the current statistics feature does not require a large amount of data, the number of method calls is not too large.
+Alternative 2 requires a change in the architecture which may result in higher complexity.
+
+=== Logging
+
+We are using `java.util.logging` package for logging. The `LogsCenter` class is used to manage the logging levels and logging destinations.
+
+* The logging level can be controlled using the `logLevel` setting in the configuration file (See <<Implementation-Configuration>>)
+* The `Logger` for a class can be obtained using `LogsCenter.getLogger(Class)` which will log messages according to the specified logging level
+* Currently log messages are output through: `Console` and to a `.log` file.
+
+*Logging Levels*
+
+* `SEVERE` : Critical problem detected which may possibly cause the termination of the application
+* `WARNING` : Can continue, but with caution
+* `INFO` : Information showing the noteworthy actions by the App
+* `FINE` : Details that is not usually noteworthy but may be useful in debugging e.g. print the actual list instead of just its size
+
+[[Implementation-Configuration]]
+=== Configuration
+
+Certain properties of the application can be controlled (e.g user prefs file location, logging level) through the configuration file (default: `config.json`).
+
+
+[[Implementation-Find]]
+=== Find feature
+
+The find feature allows the user to search for a spending based on specified field. If search results are too broad, fields may be combined to increase specificity.
+
+For example, `find n/Apple c/2.50-3.00` will find an `Apple` of cost range $`2.50` to $`3.00`.
+
+==== Implementation
+
+Find is supported by having a `Predicate` implemented for every field in Spending. Predicates are added based on valid input entered by the user.
+
+The `FindCommandParser` class stores these predicates, which are combined using Java 8 streams with an `AND` operation to form a more specific search query.
+
+The sequence diagram below demonstrates how the `find` command is executed:
+
+.Sequence diagram for an example `find` command
+image::FindSequenceDiagram.png[Find Sequence Diagram,width=75%]
+
+The following steps explain the sequence diagram:
+
+1. The user enters `find n/apple`.
+2. `LogicManager` calls `SpendingBookParser#parseCommand()`.
+3. `FindCommandParser` is created and validates user input, creating a list of `predicates`.
+4. `FindCommand` receives `predicates` and stores it in a list.
+5. On `execute()`, `predicates` are reduced and `Model#updateFilteredSpendingList(predicate)` is called to refresh the displayed list.
+
+NOTE: The lifeline for `FindCommandParser` should end at the destroy marker (X) but due to a limitation of PlantUML, the lifeline reaches the end of diagram.
+
+To summarise what happens when the user uses the `find` command, the following activity diagram is shown below:
+
+.Activity diagram for an example `find` command
+image::FindActivityDiagram.png[Find Activity Diagram,width=75%]
```

Comment:
What is List<Predicate<Spending>>? Should this be a class? Not clear what it's used for apart from being constructed.